libgstd/libgstd_json.c

```diff
@@ -0,0 +1,204 @@
+ * with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "libgstd_json.h"
+
+#include <jansson.h>
+#include <stdlib.h>
+#include <string.h>
+
+static GstdStatus gstd_json_get_value (const char *json, const char *name,
+    json_t ** root, json_t ** out);
+
+static GstdStatus
+gstd_json_get_value (const char *json, const char *name, json_t ** root,
+    json_t ** out)
+{
+  GstdStatus ret = GSTD_LIB_OK;
+  json_error_t error;
+
+  g_return_val_if_fail (json != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (name != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (root != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (out != NULL, GSTD_LIB_NULL_ARGUMENT);
+
+  *root = json_loads (json, 0, &error);
+  if (!*root) {
+    ret = GSTD_LIB_OOM;
+    goto out;
+  }
+
+  *out = json_object_get (*root, name);
+  if (!*out) {
+    ret = GSTD_LIB_NOT_FOUND;
+    json_decref (*root);
+  }
+
+out:
+  return ret;
+}
+
+GstdStatus
+gstd_json_is_null (const char *json, const char *name, int *out)
+{
+  GstdStatus ret = GSTD_LIB_OK;
+  json_t *root;
+  json_t *data;
+
+  g_return_val_if_fail (json != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (name != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (out != NULL, GSTD_LIB_NULL_ARGUMENT);
+
+  ret = gstd_json_get_value (json, name, &root, &data);
+  if (GSTD_LIB_OK != ret) {
+    goto out;
+  }
+
+  *out = json_is_null (data);
+  json_decref (root);
+
+out:
+  return ret;
+}
+
+GstdStatus
+gstd_json_get_child_char_array (const char *json,
+    const char *array_name, const char *element_name, char **out[],
+    int *array_lenght)
+{
+  GstdStatus ret = GSTD_LIB_OK;
+  json_t *root;
+  json_t *array_data;
+  json_t *data, *name;
+  json_error_t error;
+  int i, j;
+
+  g_return_val_if_fail (json != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (array_name != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (element_name != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (out != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (array_lenght != NULL, GSTD_LIB_NULL_ARGUMENT);
+
+  root = json_loads (json, 0, &error);
```

Comment:
You could check if the returned value is null and return`GSTD_LIB_BAD_PARAMETER`